internal/service/cloudsearch/domain.go

```diff
@@ -969,29 +968,28 @@ func expandIndexField(tfMap map[string]interface{}) (*cloudsearch.IndexField, er
 			options.AnalysisScheme = aws.String(analysisScheme)
 		}
 
-		if sourceFields != "" {
-			options.SourceField = aws.String(sourceFields)
-		}
-
 		if v, ok := tfMap["default_value"].(string); ok && v != "" {
 			options.DefaultValue = aws.String(v)
 		}
 
+		if v, ok := tfMap["source_fields"].(string); ok && v != "" {
+			options.SourceField = aws.String(v)
+		}
+
 		apiObject.TextOptions = options
 
 	case cloudsearch.IndexFieldTypeTextArray:
 		options := &cloudsearch.TextArrayOptions{
 			HighlightEnabled: aws.Bool(highlightEnabled),
 			ReturnEnabled:    aws.Bool(returnEnabled),
-			SourceFields:     aws.String(sourceFields),
 		}
 
 		if analysisScheme != "" {
 			options.AnalysisScheme = aws.String(analysisScheme)
 		}
 
-		if v, ok := tfMap["default_value"].(string); ok && v != "" {
-			options.DefaultValue = aws.String(v)
+		if v, ok := tfMap["source_fields"].(string); ok && v != "" {
+			options.SourceFields = aws.String(v)
 		}
```

Comment:
@ewbankkit Hi, could you please briefly explain to me why you changed this piece of code? I'm pretty new to all the stuff and it's not obvious to me...

Reply:
Just alphabetic ordering of setting the fields.

Reply:
Well, anyway I do not understand why we do not need `default_value` attribute

Reply:
Don't we just need to have separate `if` statement for `source_fields` instead of replacing `default_value` with it?

Reply:
Yeah, I did not even think about the ordering, but now I see it might be an issue with a dependency on non-existent resources.